core/src/main/scala/kafka/server/AbstractFetcherManager.scala

```diff
@@ -72,9 +74,11 @@ abstract class AbstractFetcherManager[T <: AbstractFetcherThread](val name: Stri
             initOffset = currentFetchState.fetchOffset)
           topicPartition -> initialFetchState
         }
-        addFetcherForPartitions(fetchStates)
+        allRemovedPartitionsMap ++= fetchStates
```

Comment:
Building `fetchStates` is not necessary anymore. It seems that we could replace `.map` by `.forKeyValue` and directly add partitions to `allRemovedPartitionsMap`. Would this work?